File: src/com/dafttech/terra/game/world/tiles/TileDirt.java
package com.dafttech.terra.game.world.tiles;

import com.badlogic.gdx.graphics.g2d.TextureRegion;
import com.dafttech.terra.game.world.Vector2i;
import com.dafttech.terra.resources.Resources;

public class TileDirt extends Tile implements ITileRenderBigger {
    public TileDirt() {
        super();
        setHardness(3);
    }

    @Override
    public TextureRegion getImage() {
        return Resources.TILES.getImage("dirt");
    }

    @Override
    public Vector2i getRenderSizeMultiplier() {
        return new Vector2i(2, 2);
    }
}

File: terra/src/main/scala/com/dafttech/terra/game/world/tiles/TileDirt.scala
package com.dafttech.terra.game.world.tiles

import com.badlogic.gdx.graphics.g2d.TextureRegion
import com.dafttech.terra.resources.Resources
import monix.eval.Task

class TileDirt extends Tile {
  setHardness(3)

  override val getImage: Task[TextureRegion] = Resources.TILES.getImage("dirt")
}